test/native/helpers.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react-native';
+
+export async function initializeEditor( { initialHtml } ) {
+	const Editor = require( '@wordpress/edit-post/src/editor' ).default;
+	const renderResult = render(
+		<Editor
+			postId={ 0 }
+			postType="post"
+			initialTitle="test"
+			initialHtml={ initialHtml }
+		/>
+	);
+	const { findByLabelText } = renderResult;
+	const blockListWrapper = await findByLabelText( 'block-list-wrapper' );
+	// onLayout event has to be explicitly dispatched in BlockList component,
+	// otherwise the inner blocks are not rendered.
+	fireEvent( blockListWrapper, 'layout', {
+		nativeEvent: {
+			layout: {
+				width: 100,
+			},
+		},
+	} );
+
+	const extraMethods = [
+		getByTextInAztec,
+		getBlockAtPosition,
+		addBlock,
```

Comment:
I included extra methods that will be useful for simplifying the integration tests, some of them are based on the helpers we have in the class [`EditorPage`](https://github.com/WordPress/gutenberg/blob/trunk/packages/react-native-editor/__device-tests__/pages/editor-page.js) for UI tests.

Reply:
After considering this more, traversing and searching a tree of blocks likely occurs frequently and cumbersome enough to warrant an abstraction. My only lingering question is the current approach of `getBlockAtPosition` the correct approach. 

`getBlockAtPosition` relies upon the `accessiblityLabel` to find the block. That approach will be necessary in some instances (e.g. an Image block may not have any visible text), but will defaulting to `accessibilityLabel` encourage poor practices? E.g. finding the Search block may be better tested by locating it with the "Search" label text visually rendered rather than the accessibility label. WDYT? 

For `getByTextInAztec` and `addBlock`, I still hesitate to add unused code.

Reply:
>getBlockAtPosition relies upon the accessiblityLabel to find the block. That approach will be necessary in some instances (e.g. an Image block may not have any visible text), but will defaulting to accessibilityLabel encourage poor practices? E.g. finding the Search block may be better tested by locating it with the "Search" label text visually rendered rather than the accessibility label. WDYT?

Yep, I agree that getting blocks by the accessibility label by default shouldn't be applied to all cases, as you pointed, for cases where text is present, we should locate them better by text instead. I'm thinking to rename it to `getBlockByA11yLabel` following the naming format of `react-native-testing-library`, wdyt?

>For getByTextInAztec and addBlock, I still hesitate to add unused code.

I only add them for experimenting so I'll remove them 👍 .

Reply:
> I'm thinking to rename it to `getBlockByA11yLabel` following the naming format of `react-native-testing-library`, wdyt?

Agreed. Renaming would likely improve clarity and help avoid encouraging querying elements by a11y label as the default approach. However, I begin to wonder what value this abstraction brings us. Would it be better to use RNTL's queries directly instead? 

```js
import { initializeEditor } from 'test/helpers';

// ...

// getByA11yLabel below is directly from RNTL
const { getByA11yLabel } = await initializeEditor( { initialHtml } );

expect( getByA11yLabel( /Reusable block Block. Row 1/ ) ).toBeDefined();
// ...the regex pattern of course could be simplified as needed per test...
expect( getByA11yLabel( /Reusable block Block. Row \d/ ) ).toBeDefined();
expect( getByA11yLabel( /Reusable block Block. Row/ ) ).toBeDefined();
expect( getByA11yLabel( /Reusable block/ ) ).toBeDefined();
```

I am concerned that we may be introducing a premature abstraction with our own `getBlockAtPosition`/`getBlockByA11yLabel` that provides little value. I believe the `initializeEditor` function provides considerable value, but I wonder if we do not need the rest of the utilities currently in `test/helpers`. WDYT?

Reply:
I agree, probably it's premature to add the abstraction, actually, now I noticed that we can use regular expressions so the `getBlockByA11yLabel ` function won't really add too much value. I'll remove it in favor of `getByA11yLabel` 👍.

Reply:
I've just modified this part and remove the extra functions I created to abstract it in this [commit](https://github.com/WordPress/gutenberg/pull/32038/commits/e99d7cf2a7c577596171f77b6c658ea6dd2581d8).